predicate.go

```diff
@@ -26,3 +26,16 @@ func (p *Predicate) Validate() error {
 	}
 	return nil
 }
+
+func (p *PredicateUpdateInput) MarshalJSON() ([]byte, error) {
+	if p == nil || p.Type == nil || *p.Type == "" {
+		return []byte("null"), nil
+	}
+	var predicateAsJson string
```

Comment:
I recommend a different approach,

Make a `map[string]string` from the struct and then just return the output of `json.Marshal` on that.

That way the `fmt.Sprintf` statements will not be necessary. Also, since JSON marshal usually returns JSON without any spacing I recommend that we do that too.

Reply:
Good call on the spacing. I've added that.

Calling `json.Marshal()` from within `json.Marshal()` is recursive and causes a crash. Where would `json.Marshal()` be called in this other approach?

Reply:
The code would look something like,

```
func (p *PredicateUpdateInput) MarshalJSON() ([]byte, error) {
  m := map[string]string{
    "type": *p.Type,
  }

  if p.Value != nil {
    m["value"]: *p.Value,
  }

  return JSON.Marshal(m)
}
```

There shouldn't be any recursion since we are marshalling the `map[string]string` and not the `PredicateUpdateInput`.